radiography/src/main/java/radiography/Radiography.kt

```diff
@@ -0,0 +1,221 @@
+package radiography
+
+import android.annotation.TargetApi
+import android.content.res.Resources.NotFoundException
+import android.os.Build.VERSION_CODES.CUPCAKE
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.widget.Checkable
+import android.widget.TextView
+import radiography.Radiography.scan
+
+/**
+ * Utility class to scan through a view hierarchy and pretty print it to a [String].
+ * Call [scan] or [View.scan].
+ */
+object Radiography {
+
+  /**
+   * Scans the view hierarchies and pretty print them to a [String].
+   *
+   * You should generally call this method from the main thread, as views are meant to be accessed
+   * from a single thread. If you call this from a background thread, it may work or the views
+   * might throw an exception. This method will not throw, instead the exception message will be
+   * included in the returned string.
+   *
+   * @param rootView if not null, scanning starts from [rootView] and goes down recursively (you
+   * can call the extension function [View.scan] instead). If null, scanning retrieves all windows
+   * for the current process using reflection and then scans the view hierarchy for each window.
+   *
+   * @param includeTextViewText whether to include the string content of TextView instances in
+   * the rendered view hierarchy. Defaults to false to avoid including any PII.
+   *
+   * @param textViewTextMaxLength the max size of the string content of TextView instances when
+   * [includeTextViewText] is true. When the max size is reached, the text is trimmed and
+   * ellipsized with a '…' character.
+   *
+   * @param viewFilter a filter to exclude specific views from the rendering. If a view is excluded
+   * then all of its children are excluded as well. Use [SkipIdsViewFilter] to ignore views that
+   * match specific ids (e.g. a debug drawer). Use [FocusedWindowViewFilter] to keep only the
+   * views of the currently focused window, if any.
+   */
+  @JvmStatic
+  fun scan(
+    rootView: View? = null,
+    includeTextViewText: Boolean = false,
+    textViewTextMaxLength: Int = Int.MAX_VALUE,
+    viewFilter: ViewFilter = ViewFilter.All
+  ): String = buildString {
```

Comment:
Ok so what's not there:

* Separate "go through the view hierarchy" from "here's how to render a single view" or "here's the attributes I want to collect". If we did either of these last two things, we wouldn't want includeTextViewText and textViewTextMaxLength in the API. So, solid breaking API change, so we maybe should make our mind up before release.
* Separate out "scanning a view hierarchy" from "scanning any kind of hierarchy" (e.g. Compose).  I believe this can be extracted out as a separate API  that this uses. The only question is whether "Radiography" should be for views or the more general thing and then we have a specialized version for views. Or the "specialization" is a param..

Reply:
There's an old PR that solves 1: https://github.com/square/radiography/tree/jlabanca/diagnosis
That API seems pretty flexible, we could do something like that.

I think scanning Compose hierarchy is a pretty specific and one-off use case, but that's a whole separate discussion that is out of scope for this PR.